dev-docs/creating_bindings.md

```diff
@@ -0,0 +1,12 @@
+# Creating bindings for H3
+
+H3 is a C library, in part to make it simpler to create bindings for different programming languages. Each language usually has its own way to bind to C functions, but this document can serve as a starting point and list of tips.
+
+## Function naming
+
+* The `make binding-functions` target produces a file `binding-functions` containing a list of function in the H3 public API, one per line. You can use this as part of your build process to check how much of the H3 public API your bindings expose.
```

Comment:
Maybe note somehow that the list doesn't include them memory management functions in H3 that would be useful for other non-memory managed languages, and useful for the binding code in memory managed languages to interface between the two systems.